seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java

```diff
@@ -216,4 +213,31 @@ private static SeaTunnelRow reconvert(InternalRow engineRow, SeaTunnelRowType ro
         }
         return new SeaTunnelRow(fields);
     }
+
+    private static Object reconvertArray(Object field, SeaTunnelDataType<?> dataType) {
+        BasicType<?> elementType = ((ArrayType<?, ?>) dataType).getElementType();
+        Object[] objectArray = ((ArrayData) field).toObjectArray(TypeConverterUtils.convert(elementType));
+        switch (elementType.getSqlType()) {
```

Comment:
Why we need copy data from array to another array?

Reply:
> Why we need copy data from array to another array?

Because we need to convert the object type array to the exact wrapper type array according to the SQL type. If we do not do this, when JDBC inserts the array data, it will return ClassCastException.

Reply:
But it will affects all sinks, can you handle it in clickhouse sink alone?

Reply:
I think other connectors can also work. Compared with the previous ones, they are actually the downward transformation of object arrays. The core logic has not changed.

Reply:
I have some worry about performance. Any way, the logic is fine.